src/simulation/elements/PSNS.cpp

```diff
@@ -47,8 +47,10 @@ Element_PSNS::Element_PSNS()
 //#TPT-Directive ElementHeader Element_PSNS static int update(UPDATE_FUNC_ARGS)
 int Element_PSNS::update(UPDATE_FUNC_ARGS)
 {
-	int r, rx, ry, rt;
-	if (sim->pv[y/CELL][x/CELL] > parts[i].temp-273.15f)
+	int r, rx, ry, rt, rd = parts[i].tmp2;
+	float photonWl = 0.0f;
+	bool setFilt = false;
+	if (parts[i].tmp == 0 && (sim->pv[y/CELL][x/CELL] > parts[i].temp-273.15f))
```

Comment:
I don't think this needs to be locked behind .tmp = 0. Maybe have it always sense anyway?